protocol-tests/src/main/java/fixtures/constantandclientparam/ConstantAndClientParameterServiceAsyncClient.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package fixtures.constantandclientparam;
+
+import com.azure.core.annotation.Generated;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.exception.ClientAuthenticationException;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.exception.ResourceModifiedException;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.http.rest.Response;
+import fixtures.constantandclientparam.implementation.ConstantAndClientParameterServiceClientImpl;
+import reactor.core.publisher.Mono;
+
+/** Initializes a new instance of the asynchronous ConstantAndClientParameterServiceClient type. */
+@ServiceClient(builder = ConstantAndClientParameterServiceClientBuilder.class, isAsync = true)
+public final class ConstantAndClientParameterServiceAsyncClient {
+    @Generated private final ConstantAndClientParameterServiceClientImpl serviceClient;
+
+    /**
+     * Initializes an instance of ConstantAndClientParameterServiceAsyncClient class.
+     *
+     * @param serviceClient the service client implementation.
+     */
+    @Generated
+    ConstantAndClientParameterServiceAsyncClient(ConstantAndClientParameterServiceClientImpl serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * Pass constants from the client to this function. Will pass in constant path, query, and header parameters.
+     *
+     * <p><strong>Query Parameters</strong>
+     *
+     * <table border="1">
+     *     <caption>Query Parameters</caption>
+     *     <tr><th>Name</th><th>Type</th><th>Required</th><th>Description</th></tr>
+     *     <tr><td>query-non-required-client-constant</td><td>Integer</td><td>No</td><td>Constant query property on the client that is a non-required parameter</td></tr>
+     *     <tr><td>query-non-required-constant</td><td>Integer</td><td>No</td><td>Constant query property that is a non-required parameter</td></tr>
```

Comment:
If you've set them from client, they do not need to be in javadoc. Property from builder and client should have already covered them.

But non-required-constant should not even be a constant, preprocess will transfer it to sealed choice.